core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -264,24 +291,87 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     const response = await this.threadDataBackendApiService.fetchMessagesAsync(
       threadId);
     const threadMessageBackendDicts = response.messages;
-    this.reviewMessage = threadMessageBackendDicts.map(
-      m => ThreadMessage.createFromBackendDict(m))[1].text;
+    let threadMessages = threadMessageBackendDicts.map(
+      m => ThreadMessage.createFromBackendDict(m));
+    if (threadMessages[1] !== undefined) {
+      this.reviewMessage = threadMessages[1].text;
+    }
   }
 
-  showNextItemToReview(suggestionId: string): void {
-    this.resolvedSuggestionIds.push(this.activeSuggestionId);
-    if (this.lastSuggestionToReview) {
+  gotoNextItem(): void {
+    // If the current item is the last item, do not navigate.
+    if (this.isLastItem) {
+      return;
+    }
+    // This prevents resolved contributions from getting added to the list.
+    if (!this.resolvedSuggestionIds.includes(this.activeSuggestionId)) {
+      this.skippedContributionIds.push(this.activeSuggestionId);
+    }
+
+    let lastContributionId = (
+      this.remainingContributionIds[this.remainingContributionIds.length - 1]);
+
+    this.activeSuggestionId = lastContributionId;
+    this.activeContribution = this.allContributions[
+      lastContributionId];
+
+    this.remainingContributionIds.pop();
```

Comment:
Are we going through the items from the end?

Reply:
```suggestion
    let lastContributionId = this.remainingContributionIds.pop();
    this.activeSuggestionId = lastContributionId;
    this.activeContribution = this.allContributions[
      lastContributionId];
```

Reply:
Yes we are treating these arrays as a stack.